cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+
+    Returns
+    -------
+    label_quality_score_multiannotator : pandas.DataFrame
+        pandas DataFrame in which each row corresponds to one example, with columns:
+        - ``quality_of_annotator_1``, ``quality_of_annotator_2``, ..., ``quality_of_annotator_M``: the label quality score for the labels provided by annotator M (is ``NaN`` for examples which this annotator did not label).
+        - ``consensus_labels``: the single label that is best for each example (you can control how it is derived from all annotators' labels via the argument: ``consensus_method``)
+        - ``num_annotations``: the number of annotators that have labeled each example.
+        - ``annotator_agreement``: the fraction of annotators that agree with the consensus label (only consider the annotators that labeled that particular example).
+        - ``quality_of_consensus``: label quality score for consensus label, calculated using weighted product of `label_quality_score` of `consensus_labels` and `annotator_agreement`
+        Here annotator_1, annotator_2, ..., annotator_M suffixes may be replaced column names in ``labels_multiannotator`` DataFrame used to ID the annotators.
+
+    annotator_stats : pandas.DataFrame, optional (returned if `return_annotator_stats=True`)
+        Returns overall statistics about each annotator.
+        For details, see the documentation of :py:func:`compute_multiannotator_stats<cleanlab.multiannotator.compute_multiannotator_stats>`
+    """
+
+    # Raise error if labels_multiannotator has NaN rows
+    if labels_multiannotator.isna().all(axis=1).any():
+        raise ValueError("labels_multiannotator cannot have rows with all np.NaN.")
+
+    # Raise error if labels_multiannotator has NaN columns
+    if labels_multiannotator.isna().all().any():
+        nan_columns = list(
+            labels_multiannotator.columns[labels_multiannotator.isna().all() == True]
+        )
+        raise ValueError(
+            f"""labels_multiannotator cannot have columns with all np.NaN.
+            Annotators {nan_columns} did not label any examples."""
+        )
+
+    # Raise error if labels_multiannotator has <= 1 column
+    if len(labels_multiannotator.columns) <= 1:
+        raise ValueError(
+            """labels_multiannotator must have more than one column. 
+            If there is only one annotator, use cleanlab.rank.get_label_quality_scores instead"""
+        )
+
+    # Raise error if labels_multiannotator only has 1 set of labels per example
+    if labels_multiannotator.apply(lambda s: len(s.dropna()) == 1, axis=1).all():
+        raise ValueError(
+            """Each example only has one label, collapse the labels into a 1-D array and use
+            cleanlab.rank.get_label_quality_scores instead"""
+        )
+
+    # Raise warning if no examples with 2 or more annotators agree
+    if labels_multiannotator.apply(
+        lambda s: np.array_equal(s.dropna().unique(), s.dropna()), axis=1
+    ).all():
```

Comment:
maybe we can just move this check later in the code and do it based on the `annotator_agreement` column to avoid having to do extra compute?